enhancements/opm-content-management.md

```diff
@@ -0,0 +1,147 @@
+
+A new sub-command `graph` will be added under `opm index` to allow operator authors or cluster admins to understand the update graph for a given Operator in a catalog, so that the update graph can be reasoned about for correctness and available versions.
+
+The `graph` sub-command will have a `type` flag, to denote the type of graph visualization, one of `ascii` (for ASCII representation of the graph), `graphical` (for a graphical visualization of the graph, possibly using (go-graphviz)[https://github.com/goccy/go-graphviz] that will ultimately be exported to a png file in the working directory), or `json` (for a human readable data format of the graph). The default option will be `ascii`. 
+
+For example, `opm index graph --type=graphical --package=example-operator quay.io/example-namespace/example-index:v0.0.1`
+
+### Support for deamonless bundle validation
+
+Currently, `opm` requires the docker deamon for bundle validation (mostly for unpacking container image) by default. This dependency on a deamon will be removed by using containerd or podman as default container tools to pull and unpack container images for validating the bundle.    
+
+As an operator author, I want to be able to validate bundle image without a daemon or source out to a binary/CLI tool in local host. Ideally, bundle validation tool is a self-container tool without requiring any external tool being installed.
+
+### Support for removing an individual Operator bundle from an index
+
+Currently, `opm` has an `index rm` option that allows for the deletion of a package entirely from an index. 
+
+`opm` will be enhanced to provide an option to delete an operator from a package:
+a) From the leaf end of the update graph for a package 
+b) From the beginning of the update graph for a package 
+c) From the middle of an update graph for a package
+
+To achieve this, two flags will be introduced under the `index rm` sub-command, `package` and `bundle`.
+
+`opm index rm --package=<package-name> quay.io/example-namespace/example-index:v0.0.1` will remove a package from an index 
+
+`opm index rm --bundle=<bundle-path> quay.io/example-namespace/example-index:v0.0.1` will remove the bundle from a package in an index. 
+
+### Support for removing a channel from a package in a bundle 
+
+`opm index rm` will also have a `channel` flag, that will be paired with the `package` flag to remove a channel from a package, and all the bundles that belong to that channel. 
+
+```
+$ opm index rm --channel=beta --package=example-operator quay.io/example-namespace/example-indexv-0.0.1
+
+Removed bundles example-operatorv0.0.1, example-operatorv0.0.2 from package example-operator. 
+Deleted channel beta from example-operator.
+
+``` 
+
+#### Open question
+
+Once a bundle is removed from a pacakge in an index, is the onus on `opm` to handle the upgrade of bundles that have already been installed in a cluster, but have been removed from the index?  
```

Comment:
I think we do want OLM to manage the `skip` list for surrounding operator updates, so that clusters that have seen the "bad" operator can update properly. See: https://github.com/operator-framework/operator-lifecycle-manager/blob/master/doc/design/how-to-update-operators.md#skipping-updates